internal/webservice/webservice.go

```diff
@@ -47,6 +54,8 @@ func (ws *WebService) Start(ctx context.Context) error {
 
 	ws.initRoutes(ctx)
 
+	ws.setupMiddlewares()
```

Comment:
Wouldn't it be better if middlewares are created within `initRoutes`? That way, initRoutes can install all handlers and middlewares that it needs. After the `initRoutes` call, you know that routes are created and appropriate middlewares are installed. This doesn't need to be a two-step process :relaxed:

Reply:
Thanks, I was thinking about and could not decide which one would be better. Will update

Reply:
the method could also be rename `initRoutes` -> `init`